cordova-lib/src/plugman/platforms/ios.js

```diff
@@ -184,13 +192,76 @@ module.exports = {
 
 
         var xcBuildConfiguration = xcodeproj.pbxXCBuildConfigurationSection();
-        var plist_file_entry = _.find(xcBuildConfiguration, function (entry) { return entry.buildSettings && entry.buildSettings.INFOPLIST_FILE; });
-        var plist_file = path.join(project_dir, plist_file_entry.buildSettings.INFOPLIST_FILE.replace(/^"(.*)"$/g, '$1').replace(/\\&/g, '&'));
-        var config_file = path.join(path.dirname(plist_file), 'config.xml');
 
-        if (!fs.existsSync(plist_file) || !fs.existsSync(config_file)) {
+        // CB-9033
+        var plist_file_index;
+        var plist_file_entry = _.find(xcBuildConfiguration, function (entry,index) { 
+            if (entry.buildSettings && entry.buildSettings.INFOPLIST_FILE) { 
+
+                var plist_file = path.join(project_dir, entry.buildSettings.INFOPLIST_FILE.replace(/^"(.*)"$/g, '$1').replace(/\\&/g, '&'));
+                 if (!fs.existsSync(plist_file)) 
+                    return false; 
+                
+                var config_file = path.join(path.dirname(plist_file), 'config.xml'); 
+                  if (!fs.existsSync(config_file)) 
+                    return false; 
+
+                // only return project that contains both a plist and a config.xml (especially to discard apple watch extension/app)
+                plist_file_index = index;
```

Comment:
Instead of checking for config.xml can we check for SKIP_INSTALL property in BuildSettings. SKIP_INSTALL property should be missing or equals to NO for the iPhone app target and should equals to YES for WatchKit Extension and WatchKit App.